sdk/src/trace/tracer.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "opentelemetry/trace/tracer.h"
+#include "src/trace/recorder.h"
+
+#include <memory>
+
+namespace opentelemetry
+{
+namespace sdk
+{
+namespace trace
+{
+class Tracer final : public trace_api::Tracer, public std::enable_shared_from_this<Tracer>
+{
+public:
+  explicit Tracer(std::unique_ptr<Recorder> &&recorder) noexcept : recorder_{std::move(recorder)} {}
+
+  Recorder &recorder() const noexcept { return *recorder_; }
```

Comment:
Implies the recorder passed to the constructor can't be nullptr, right? Is this worth documenting?

Reply:
Yes, I added a note.